core/src/main/scala/za/co/absa/spline/core/listener/SplineQueryExecutionListener.scala

```diff
@@ -51,6 +58,14 @@ class SplineQueryExecutionListener(harvesterFactory: DataLineageBuilderFactory,
     }
   }
 
+  def wasResultIgnored(lineage: DataLineage): Boolean =
+    lineage.rootOperation match {
+      case op.Write(_, _, _, _, writeMetrics, readMetrics) =>
+        writeMetrics get "numFiles" exists 0.==
+      case _ =>
+        sys.error(s"Unexpected root operation: ${lineage.rootOperation.getClass}")
```

Comment:
Why not rather use "warning" or throw an exception? Error usually means that the program cannot continue.

Reply:
Because the program cannot continue if this happens :) This code expects the root operation to always be `Write`. Basically I could safely remove `case _`  block and it won't change anything. I have put it there to emphasize and describe this assumption. Another way to do it would be `assume(rootOperation.isInstanceOf[Write])`, but `case _ => sys.error` reads better to me.

Reply:
In my other projects I usually had a common helper object 'Assert` with methods like `shouldNeverReachHere()`, `shouldNeverBeThrown(ex)` etc that would simply throw a runtime exception with the according description.

Reply:
Oh, sys.error actually throws the RuntimeException.